h2o_benchamarks.ipynb

```diff
@@ -49,41 +50,160 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "44109817-9ae7-49bb-9fff-ff791757e174",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ds = \"5GB\" # choose \"05GB\" , \"5GB\" or \"50GB\"\n",
+    "report_dir = \"performance-reports-2\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dee138b7-47e4-4c6b-97d6-28fdc3f8a2ad",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ddf = dd.read_parquet(\n",
+    "    data_size[ds],\n",
+    "    engine=\"pyarrow\",\n",
+    "    storage_options={\"anon\": True},\n",
+    ")\n",
+    "ddf"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8881a3d4-645a-43f1-8b2c-35a561b1a50b",
+   "metadata": {},
+   "source": [
+    "### Memory usage and `dtype` inspection"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b131f692-5aaa-48c2-8123-d9abe7e50d0f",
+   "metadata": {},
+   "source": [
+    "Total size on disk and in memory"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "a8cea0dc",
+   "id": "fcc80a57-7c97-4625-a180-bab94fc5b54b",
    "metadata": {},
    "outputs": [],
    "source": [
-    "def read_data(data_uri):\n",
-    "    ddf = dd.read_parquet(\n",
-    "                data_uri, engine=\"pyarrow\", storage_options={\"anon\": True}\n",
-    "            )\n",
-    "    return ddf"
+    "from dask.utils import format_bytes\n",
+    "import s3fs\n",
+    "\n",
+    "fs = s3fs.S3FileSystem()\n",
+    "disk_size = fs.du(os.path.dirname(data_size[ds]))\n",
+    "format_bytes(disk_size)"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "27aaa620",
+   "id": "ff48d21b-eb80-42ab-a7e9-88290f72cd06",
    "metadata": {},
    "outputs": [],
    "source": [
-    "ds = \"50GB\" # choose \"05GB\" , \"5GB\" or \"50GB\"\n",
-    "ddf = read_data(data_size[ds])"
+    "in_memory_size = ddf.memory_usage(deep=True).sum().compute()\n",
+    "format_bytes(in_memory_size)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0d952df8-4987-4f09-a8d5-35e99cf77894",
+   "metadata": {},
+   "source": [
+    "Data is ~4x larger in memory than on disk"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f9810641-b89c-4d8d-a480-42efed2d214a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ddf.memory_usage_per_partition(deep=True).compute() / 1e6"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f58d4d28-18bf-434b-93d8-a8c1306af1aa",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ddf.astype({\"id3\": \"string[pyarrow]\"}).memory_usage_per_partition(deep=True).compute() / 1e6"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b27cafd0-5384-4347-9fda-9c12f8cb7cb0",
+   "metadata": {},
+   "source": [
+    "Column `id3` takes up the most memory (by far). This is a `categorical` with lots of categories. Using `string[pyarrow]` instead of `category` takes up ~2x memory."
```

Comment:
I'm a bit confused here, takes up ~2x more or less memory. I assume is less? Is this also in disk or in memory?

Reply:
This is because for the high-cardinality case, each partition of the categorical column basically carries the footprint of the entire column. Poking at this dataset is what prompted https://github.com/dask/dask/issues/9392

I'd say the behavior here is pathological enough that we should rewrite the whole dataset to remove categoricals.

Reply:
Agreed. This is just meant as a small comment with some observations.

Reply:
That's a good point. Here are the original dtypes in R for the h2o data. If we can be consistent with it  and not use categories then we should. At the moment I was adapting a script provided by Powers, and I assume it was ok, but it looks like we can just use strings.